classwork/classwork3.2/src/tests.cpp

```diff
@@ -0,0 +1,101 @@
+#include <iostream>
+#include <cmath>
+#include "Triangle.h"
+#include "Color.h"
+#include "UnitTests.h"
+
+int main() {
+    UnitTests testSuite;
+
+    testSuite.addTest("TestArea_3_4_5",
+                      []() {
+                          Triangle triangle(3, 4, 5);
+                          double s = (3 + 4 + 5) / 2.0;
+                          double expected = std::sqrt(s * (s - 3) * (s - 4) * (s - 5));
+                          ASSERT_EQ(triangle.area(), expected);
+    });
+
+    testSuite.addTest("TestArea_1_1_1",
+                      []() {
+                          Triangle triangle(1, 1, 1);
+                          double s = (1 + 1 + 1) / 2.0;
+                          double expected = std::sqrt(s * (s - 1) * (s - 1) * (s - 1));
+                          ASSERT_EQ(triangle.area(), expected);
+    });
+
+    testSuite.addTest("TestArea_5_12_13",
+                      []() {
+                          Triangle triangle(5, 12, 13);
+                          double s = (5 + 12 + 13) / 2.0;
+                          double expected = std::sqrt(s * (s - 5) * (s - 12) * (s - 13));
+                          ASSERT_EQ(triangle.area(), expected);
+    });
+
+    testSuite.addTest("TestNegativeSide",
+                      []() {
+                          Triangle triangle(-3, 4, 5);
+                          ASSERT_EQ(triangle.area(), -1);
+    });
+
+    testSuite.addTest("TestZeroSide",
+                      []() {
+                          Triangle triangle(0, 4, 5);
+                          ASSERT_EQ(triangle.area(), -1);
+    });
+
+    testSuite.addTest("TestInvalidTriangle",
+                      []() {
+                          Triangle triangle(1, 2, 3);
+                          ASSERT_EQ(triangle.area(), -1);
+    });
+
+    testSuite.addTest("TestColor_100_50_25",
+                      []() {
+                            Color color(100, 50, 25);
+                            Color magic = color.magicColor();
+                            auto [r, g, b] = magic.getRGB();
+                            ASSERT_EQ(r, 49);  // (100 / 2) - 1 = 49
+                            ASSERT_EQ(g, 98);  // (50 * 2) - 2 = 98
+                            ASSERT_EQ(b, 25);  // Unchanged
+    });
+
+    testSuite.addTest("TestColor_0_0_0",
+                      []() {
+                            Color color(0, 0, 0);
+                            Color magic = color.magicColor();
```

Comment:
minor: we can use empty lines between logical blocks(build-operate-check or arrange/act/assert) of unit test to visually simplify reading